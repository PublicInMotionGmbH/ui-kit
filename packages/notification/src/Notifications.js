import React from 'react'
import PropTypes from 'prop-types'
import { CSSTransition, TransitionGroup } from 'react-transition-group'
import { prefix } from '@talixo/commons'

const name = prefix('notifications')

const Fade = ({ children, ...props }) => (
  <CSSTransition {...props} timeout={600} classNames='fade'>
    {children}
  </CSSTransition>
)

/**
 * Component which represents Notifications.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {string} [props.style]
 * @returns {React.Element}
 */
class Notifications extends React.Component {
  constructor () {
    super()
    this.state = {
      items: []
    }
  }

  componentDidMount () {
    setTimeout(() => this.setState({items: Array.isArray(this.props.children) ? this.props.children : [this.props.children]}), 1000)
  }

  handleRemove (i) {
    let newItems = this.state.items.slice()
    newItems.splice(i, 1)
    this.setState({ items: newItems })
  }

  render () {
    const { className, style } = this.props
    return (
      <div className={`${name}-container`}>
        <div className={`${name}-wrapper`}>
          <TransitionGroup className={name}>
            {this.state.items.map((child, i) => (
              <Fade key={child.props.id}>
                {React.cloneElement(child, {
                  className,
                  style,
                  handleRemove: () => this.handleRemove(i)
                })}
              </Fade>
            ))}
          </TransitionGroup>
        </div>
      </div>
    )
  }
}

Notifications.propTypes = {
  /** Additional class name passed to notifications */
  className: PropTypes.string,

  /** Additional styles passed to notifications */
  style: PropTypes.object
}

Notifications.defaultProps = {
}

export default Notifications
