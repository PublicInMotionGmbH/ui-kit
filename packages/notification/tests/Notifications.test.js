import React from 'react'
import Notifications from '../src/Notifications'
import { shallow } from 'enzyme'

describe('<Notifications />', () => {
  it('renders correctly one child', () => {
    const wrapper = shallow(
      <Notifications>
        <div id={0}>Notification</div>
      </Notifications>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly children', () => {
    const wrapper = shallow(
      <Notifications>
        <div id={0}>Notification 1</div>
        <div id={1}>Notification 2</div>
        <div id={2}>Notification 3</div>
      </Notifications>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders .left correctly', () => {
    const wrapper = shallow(
      <Notifications className='left'>
        <div id={0}>Notification</div>
      </Notifications>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
