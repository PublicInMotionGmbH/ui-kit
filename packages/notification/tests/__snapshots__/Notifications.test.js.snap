// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notifications /> renders .left correctly 1`] = `
<div
  className="talixo-notifications-container"
>
  <div
    className="talixo-notifications-wrapper"
  >
    <TransitionGroup
      childFactory={[Function]}
      className="talixo-notifications"
      component="div"
    />
  </div>
</div>
`;

exports[`<Notifications /> renders correctly children 1`] = `
<div
  className="talixo-notifications-container"
>
  <div
    className="talixo-notifications-wrapper"
  >
    <TransitionGroup
      childFactory={[Function]}
      className="talixo-notifications"
      component="div"
    />
  </div>
</div>
`;

exports[`<Notifications /> renders correctly one child 1`] = `
<div
  className="talixo-notifications-container"
>
  <div
    className="talixo-notifications-wrapper"
  >
    <TransitionGroup
      childFactory={[Function]}
      className="talixo-notifications"
      component="div"
    />
  </div>
</div>
`;
